### ISSUE TITLE
KP with combination Ice Lake + Big Sur + AppleALC (alc230)

### ISSUE THREAD

**Issue by kalkmann** [acidanthera/bugtracker#1447]
Hello, I would like to report a recurring Kernel Panic on Ice Lake + Big Sur + AppleALC platforms

In addition to many reported cases of KP on hardware similar to my configuration (Lenovo S145 i7 1065G7 + alc230), the same panic has already been reported on laptops of different brand, processor and codec, such as the Acer a514-53 with 1035G1 CPU and alc256 codec.

Apparently the Kernel Panic is related to the microphone and is easily reproducible trying to use Siri, or VoiceOver in the keyboard settings, for example, but also experimented many random ones, browsing multimedia sites, regardless of the browser.

Using exactly the same EFI, Kexts, the same existing layouts (20 or 13), there is no problem in Catalina 10.15.7.

Sorry if I didn't understand the process of reporting issues. Thanks in advance.

Here is the KP report in Big Sur 11.1:
```
panic(cpu 4 caller 0xffffff80023ed0e6): Kernel trap at 0xffffff80055142e9, type 6=invalid opcode, registers:
CR0: 0x0000000080010033, CR2: 0x000000010fcfe000, CR3: 0x000000027bb74012, CR4: 0x00000000003626e0
RAX: 0xffffffb06a8c7000, RBX: 0xffffff93567a3100, RCX: 0x0000000000000000, RDX: 0x0000000000000000
RSP: 0xffffffb06cfbbb38, RBP: 0xffffffb06cfbbb50, RSI: 0x0000000000000001, RDI: 0xffffffb06c456000
R8:  0x0000000000000001, R9:  0x0000000000000470, R10: 0x0000000000000020, R11: 0x0000000000000020
R12: 0xffffffb06c456000, R13: 0xffffff9355c20be0, R14: 0x0000000000000470, R15: 0xffffffb06a8c7000
RFL: 0x0000000000010206, RIP: 0xffffff80055142e9, CS:  0x0000000000000008, SS:  0x0000000000000010
Fault CR2: 0x000000010fcfe000, Error code: 0x0000000000000000, Fault CPU: 0x4, PL: 0, VF: 0

Backtrace (CPU 4), Frame : Return Address
0xffffffb06cfbb550 : 0xffffff80022b9aed mach_kernel : _handle_debugger_trap + 0x3dd
0xffffffb06cfbb5a0 : 0xffffff80023fc6e3 mach_kernel : _kdp_i386_trap + 0x143
0xffffffb06cfbb5e0 : 0xffffff80023ecd1a mach_kernel : _kernel_trap + 0x55a
0xffffffb06cfbb630 : 0xffffff800225ea2f mach_kernel : _return_from_trap + 0xff
0xffffffb06cfbb650 : 0xffffff80022b938d mach_kernel : _DebuggerTrapWithState + 0xad
0xffffffb06cfbb770 : 0xffffff80022b9678 mach_kernel : _panic_trap_to_debugger + 0x268
0xffffffb06cfbb7e0 : 0xffffff8002abe3ca mach_kernel : _panic + 0x54
0xffffffb06cfbb850 : 0xffffff80023ed0e6 mach_kernel : _sync_iss_to_iks + 0x2c6
0xffffffb06cfbb9d0 : 0xffffff80023ecdcd mach_kernel : _kernel_trap + 0x60d
0xffffffb06cfbba20 : 0xffffff800225ea2f mach_kernel : _return_from_trap + 0xff
0xffffffb06cfbba40 : 0xffffff80055142e9 com.apple.vecLib.kext : _vDSP_zvmul + 0xed99
0xffffffb06cfbbb50 : 0xffffff7fa137927e com.apple.driver.DspFuncLib : __ZN23DspFuncClientGainAdjust7processEjj + 0xfc
0xffffffb06cfbbb90 : 0xffffff7fa13348e6 com.apple.driver.DspFuncLib : __ZN14DspFuncManager7processEPKvjS1_jjjjj + 0x188
0xffffffb06cfbbc40 : 0xffffff7fa1251e34 com.apple.driver.AppleHDA : __ZN14AppleHDAEngine14dspOrCopyAudioEPvjS0_jjjjj + 0x88
0xffffffb06cfbbca0 : 0xffffff7fa1252a15 com.apple.driver.AppleHDA : __ZN14AppleHDAEngine37convertInputSamples_StereoToStereo_32EPKvPvjjPK20_IOAudioStreamFormatP13IOAudioStream + 0x47f
0xffffffb06cfbbd50 : 0xffffff7fa1251cfe com.apple.driver.AppleHDA : __ZN14AppleHDAEngine34convertInputSamples_StereoToStereoEPKvPvjjPK20_IOAudioStreamFormatP13IOAudioStream + 0x14
0xffffffb06cfbbd70 : 0xffffff7fa12523b4 com.apple.driver.AppleHDA : __ZN14AppleHDAEngine19convertInputSamplesEPKvPvjjPK20_IOAudioStreamFormatP13IOAudioStream + 0x6c
0xffffffb06cfbbd90 : 0xffffff800474d9d6 com.apple.iokit.IOAudioFamily : __ZN13IOAudioStream16readInputSamplesEP19IOAudioClientBufferj + 0x11c
0xffffffb06cfbbdf0 : 0xffffff8004748908 com.apple.iokit.IOAudioFamily : __ZN23IOAudioEngineUserClient18performClientInputEjP22IOAudioClientBufferSet + 0x142
0xffffffb06cfbbe60 : 0xffffff8004748328 com.apple.iokit.IOAudioFamily : __ZN23IOAudioEngineUserClient15performClientIOEjjbjjj + 0x14e
0xffffffb06cfbbeb0 : 0xffffff8002a5156e mach_kernel : _iokit_user_client_trap + 0x18e
0xffffffb06cfbbef0 : 0xffffff80023d0ec3 mach_kernel : _mach_call_munger64 + 0x273
0xffffffb06cfbbfa0 : 0xffffff800225f216 mach_kernel : _hndl_mach_scall64 + 0x16
      Kernel Extensions in backtrace:
         com.apple.vecLib.kext(1.2)[90633E0B-7D0C-3DDC-8234-E6DE6E3F8F37]@0xffffff80054c1000->0xffffff800556bfff
            dependency: com.apple.kec.Libm(1)[18ECED72-27A2-3946-9D28-FD1AAE3EE8DB]@0xffffff80051f3000->0xffffff8005200fff
         com.apple.iokit.IOAudioFamily(300.6.1)[5F231B4B-7E8C-3F8A-844B-CE489D7529C0]@0xffffff800473a000->0xffffff8004757fff
            dependency: com.apple.vecLib.kext(1.2.0)[90633E0B-7D0C-3DDC-8234-E6DE6E3F8F37]@0xffffff80054c1000->0xffffff800556bfff
         com.apple.driver.DspFuncLib(283.15)[B41F8C3E-D31F-3A51-8D79-CD9AA21C1DE3]@0xffffff7fa1312000->0xffffff7fa13f4fff
            dependency: com.apple.driver.AppleEFINVRAM(2.1)[E4DF1D8B-0DA9-33A8-A7B6-17DCA1678FC0]@0xffffff8003659000->0xffffff8003662fff
            dependency: com.apple.iokit.IOAudioFamily(300.6.1)[5F231B4B-7E8C-3F8A-844B-CE489D7529C0]@0xffffff800473a000->0xffffff8004757fff
            dependency: com.apple.kext.OSvKernDSPLib(529)[8F4BD574-40D0-339E-8F82-17109DD9E7E4]@0xffffff7fa31fa000->0xffffff7fa3208fff
            dependency: com.apple.vecLib.kext(1.2.0)[90633E0B-7D0C-3DDC-8234-E6DE6E3F8F37]@0xffffff80054c1000->0xffffff800556bfff
         com.apple.driver.AppleHDA(283.15)[52079101-0266-3BF6-B203-88CD9D4E8321]@0xffffff7fa1230000->0xffffff7fa12bffff
            dependency: com.apple.driver.AppleHDAController(283.15)[242DDD3E-CAD8-38E4-B4B2-088E89A06AED]@0xffffff7fa12e5000->0xffffff7fa12f7fff
            dependency: com.apple.driver.DspFuncLib(283.15)[B41F8C3E-D31F-3A51-8D79-CD9AA21C1DE3]@0xffffff7fa1312000->0xffffff7fa13f4fff
            dependency: com.apple.iokit.IOAudioFamily(300.6.1)[5F231B4B-7E8C-3F8A-844B-CE489D7529C0]@0xffffff800473a000->0xffffff8004757fff
            dependency: com.apple.iokit.IOGraphicsFamily(585)[010A56CD-1229-3DA1-B6A8-666F37B48F21]@0xffffff7fa2a61000->0xffffff7fa2a8ffff
            dependency: com.apple.iokit.IOHDAFamily(283.15)[C1B89A4F-D2CA-3052-98E5-6855B8A5F359]@0xffffff7fa1405000->0xffffff7fa1408fff
            dependency: com.apple.iokit.IONDRVSupport(585)[A04873E0-470E-3D5B-B64F-51467022E77A]@0xffffff7fa2aa6000->0xffffff7fa2aaefff

Process name corresponding to current thread: coreaudiod
Boot args: -v alcid=20 watchdog=0 -igfxblr -igfxnohdmi -noDC9 igfxfw=2 keepsyms=1 debug=0x100 chunklist-security-epoch=0 -chunklist-no-rev2-dev

Mac OS version:
20C69

Kernel version:
Darwin Kernel Version 20.2.0: Wed Dec  2 20:39:59 PST 2020; root:xnu-7195.60.75~1/RELEASE_X86_64
Kernel UUID: 82E2050C-5936-3D24-AD3B-EC4EC5C09E11
KernelCache slide: 0x0000000002000000
KernelCache base:  0xffffff8002200000
Kernel slide:      0x0000000002010000
Kernel text base:  0xffffff8002210000
__HIB  text base: 0xffffff8002100000
System model name: MacBookPro16,2 (Mac-5F9802EFE386AA28)
System shutdown begun: NO
Panic diags file available: YES (0x0)
Hibernation exit count: 0

System uptime in nanoseconds: 77204405708
Last Sleep:           absolute           base_tsc          base_nano
  Uptime  : 0x00000011f9bdc8f1
  Sleep   : 0x0000000000000000 0x0000000000000000 0x0000000000000000
  Wake    : 0x0000000000000000 0x000000028bedd5d1 0x0000000000000000
```

**Comment by vit9696** [comment]
This looks like a resource incompatibility to me with your codec. Not much we can do on our side, you need to create custom audio resources for your own hardware.

**Comment by kalkmann** [comment]
Hello, thanks for your attention. I managed to find a solution, compiling the extension and creating a new layout. **In this new layout I deleted the signal processing dictionaries on the audio inputs**.

As I mentioned, the same problem with audio inputs on Big Sur is being reported:

Acer with 1035g1 and alc256 CPU:
https://www.olarila.com/topic/9918-olarila-hackbook-lenovo-ideapad-s145-10th-gen-catalina-big-sur-full-dsdt-patches-clover-and-opencore/?do=findComment&comment = 137180

The same Panic on a desktop (!):
https://www.insanelymac.com/forum/topic/346064-kernel-panic-when-use-microphone/

Another, which he solved using VoodooHDA:
https://www.olarila.com/topic/9918-olarila-hackbook-lenovo-ideapad-s145-10th-gen-catalina-big-sur-full-dsdt-patches-clover-and-opencore/?do=findComment&comment = 137812

Attached is the layout created here to solve on my Lenovo Laptop with 1065G7 CPU and ALC230 codec (I only copied and edited the existing layout 20).

I hope I have helped in some way.

Sincerely,

Juliano Kalkmann

[layout22.xml.zip](https://github.com/acidanthera/bugtracker/files/5849126/layout22.xml.zip)
